test/integration/scheduler/volume_binding_test.go

```diff
@@ -246,9 +255,94 @@ func TestVolumeBinding(t *testing.T) {
 	}
 }
 
+// TestVolumeBindingRescheduling tests scheduler will retry scheduling when needed.
+func TestVolumeBindingRescheduling(t *testing.T) {
+	features := map[string]bool{
+		"VolumeScheduling":              true,
+		"PersistentLocalVolumes":        true,
+		"DynamicProvisioningScheduling": true,
+	}
+	config := setupCluster(t, "volume-scheduling", 2, features)
+	defer config.teardown()
+
+	storageClassName := "local-storage"
+
+	cases := map[string]struct {
+		pod        *v1.Pod
+		pvcs       []*testPVC
+		trigger    func(config *testConfig)
+		shouldFail bool
+	}{
+		"reschedule on WaitForFirstConsumer storage class added": {
```

Comment:
For static binding, you could also have this scenario where:
- create pvc, there are no pvs available
- create storageclass
- create pvs

Reply:
@msau42  updated in https://github.com/kubernetes/kubernetes/pull/65616/commits/8eb7a3b945f4db77a938f310921ee9c80fff169f

In testing this scenario, I found we need to move pods into active queue on PV add & update:

- on PV add

make sure unbounded PVs created for static provisioning and delay binding storage class will reschedule stuck pods 
- on PV update

IIUC, `Scheduler.bindVolumesWorker` needs to send back pod to scheduler always, so it fails with error in all cases, but [error handler only puts pod back to unschedule queue on error](https://github.com/kubernetes/kubernetes/blob/v1.12.0-alpha.0/pkg/scheduler/factory/factory.go#L1340). If `Scheduler.bindVolumesWorker` failed to update PV due to conflicts when PV is modified by others, PVC will not be bound to volume by PV controller, so [no PVC update event to move pods to active queue](https://github.com/kubernetes/kubernetes/blob/v1.12.0-alpha.0/pkg/scheduler/factory/factory.go#L486). In this case, we need to move pods to active queue on PV update event.